lib/src/main/kotlin/com/swmansion/starknet/provider/rpc/JsonRpcProvider.kt

```diff
@@ -68,6 +66,28 @@ class JsonRpcProvider(
         return HttpRequest(payload, buildJsonHttpDeserializer(responseSerializer, deserializationJson), httpService)
     }
 
+    fun <T> combineBatchRequest(
+        calls: List<Request<T>>,
+    ): BatchHttpRequest<T> {
+        val httpRequests = calls.map { it as HttpRequest<T> }
+        val updatedJsonObjects = httpRequests.mapIndexed { index, request ->
+            val jsonMap = Json.parseToJsonElement(request.payload.body.toString()).jsonObject.toMutableMap()
+            jsonMap["id"] = JsonPrimitive(index)
+            JsonObject(jsonMap)
+        }
+
+        val jsonArray = JsonArray(updatedJsonObjects)
+        val deserializer = httpRequests.first().deserializer
+        val payload = HttpService.Payload(
+            url,
+            "POST",
+            emptyList(),
+            jsonArray.toString(),
```

Comment:
we first made if json object, and then converted to string. I wonder if there is a way to make it more efficient. this might be a nitpick.

Reply:
I'm not sure there's much room for improvement. We use the same approach in original `buildRequest`. 

We can use `Json.encodeToString` on `List<JsonRpcRequest>` if we opt for the approach I suggested above, but I doubt there would be much difference anyway.

Reply:
Yes, calling `joinToString` on `updatedJsonObjects` with specifying `prefix` and `postifx` params is handy here, updated 👍

Reply:
Ok, can we leave it then with the original implementation?